index.bs

```diff
@@ -3726,6 +3788,38 @@ SHOULD be aborted.
     See [WHATWG HTML WG Issue #2711](https://github.com/whatwg/html/issues/2711) for more details.
 
 
+## WebAuthn Interfaces ## {#sctn-interfaces}
+
+The subection below defines custom interfaces used throughout WebAuthn.
+
+### Custom WebAuthn Exceptions ### {#iface-custom-webauthn-exceptions}
+
+For descriptions of these exceptions,
+please see [[#sctn-create-request-exceptions]] and [[#sctn-get-request-exceptions]].
+
+<xmp class="idl">
+[Exposed=Window, Serializable]
+interface HybridPrerequisitesError : DOMException {
+};
```

Comment:
I think all of these [need to have explicit constructor operations and (de)serialization steps](https://webidl.spec.whatwg.org/#idl-DOMException-derived-interfaces):

>- The [interface](https://webidl.spec.whatwg.org/#dfn-interface) must have a [constructor operation](https://webidl.spec.whatwg.org/#idl-constructors) which sets the instance’s [name](https://webidl.spec.whatwg.org/#domexception-name) to the interface’s [identifier](https://webidl.spec.whatwg.org/#dfn-identifier).
>- Their [constructor operation](https://webidl.spec.whatwg.org/#idl-constructors) must take as its first parameter an [optional](https://webidl.spec.whatwg.org/#dfn-optional-argument) [DOMString](https://webidl.spec.whatwg.org/#idl-DOMString) named message defaulting to the empty string, and must set the instance’s [message](https://webidl.spec.whatwg.org/#domexception-message) to message.
- [...]
- They should be [serializable objects](https://html.spec.whatwg.org/multipage/structured-data.html#serializable-objects), whose [serialization steps](https://html.spec.whatwg.org/multipage/structured-data.html#serialization-steps) and [deserialization steps](https://html.spec.whatwg.org/multipage/structured-data.html#deserialization-steps) preserve the additional information.

It doesn't look like those can be inherited. So this would need to be something like this:

```suggestion
interface HybridPrerequisitesError : DOMException {
    constructor(optional DOMString message = "");
};

The <code>new HybridPrerequisitesError(|message|)</code> constructor steps are to set [=this=]'s {{DOMException/name}} to `"HybridPrerequisitesError"`.

{{HybridPrerequisitesError}} objects are [=serializable objects=].

Their [=serialization steps=], given |value| and |serialized|, are to run the {{DOMException}} [=serialization steps=] given |value| and |serialized|.

Their [=deserialization steps=], given |serialized| and |value|, are to run the {{DOMException}} [=deserialization steps=] given |serialized| and |value|.
```

...and repeat that for each derived interface. :face_with_diagonal_mouth:

Reply:
Oh, then I misread [the example](https://webidl.spec.whatwg.org/#example-domexception-derived-interface) - I thought it was saying that stuff came along for free, not that it was spelling out exactly what had to be written out as per the steps you highlighted. That's...suboptimal, but if it's the reality then I suppose I'll expand all this out.

Reply:
@emlun I've made these changes in a way that I think satisfies all of what https://webidl.spec.whatwg.org/#idl-DOMException-derived-interfaces says about adding custom DOMErrors:

![Screenshot 2024-08-28 at 11 02 06 AM](https://github.com/user-attachments/assets/fe9ab59d-c166-4e19-9c84-7701f15a928f)

![Screenshot 2024-08-28 at 11 02 11 AM](https://github.com/user-attachments/assets/51bfb6b9-e608-454e-a1b5-03e02bf3d199)